site/docs/v1/tech/lambdas/testing-lambdas.md

```diff
@@ -0,0 +1,380 @@
+
+Let's take an example where you check if users have email addresses from a country sanctioned by the United States, such as North Korea or Cuba. You call the external site `https://issanctioned.example.com` with an email address, and you're told whether or not the domain is banned.
+
+Add this new function to `test.js`. 
+
+```js
+{% remote_include 'https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/documentation_snippets/test_3.js' %}
+```
+
+This is the lambda function that would run on FusionAuth, similar to the "Hello World" function described earlier.
+
+Now add the two tests below. The first test checks if North Korea (`.kp`) is banned and the second if Canada (`.ca`) is allowed.
+
+```js
+{% remote_include 'https://raw.githubusercontent.com/FusionAuth/fusionauth-example-testing-lambdas/main/complete-application/documentation_snippets/test_4.js' %}
+```
+
+This test function uses `fetchMock` to mock the external service that would be called from the lambda function in FusionAuth. The mocks for the JWT, user, and registration objects are all simple `{}` objects you can pass as parameters to the `populate()` lambda.
+
+Finally, run the tests.
+
+```bash
+node test.js  | npx faucet
+```
+
+If all your unit tests for a lambda pass, you can safely upload it to FusionAuth manually or with the CLI for further testing.
+
+If your HTTP Connect fetch request fails when deployed to FusionAuth, please review the [documentation](/docs/v1/tech/lambdas/#using-lambda-http-connect). In particular, ensure you are using a license and have purchased the correct plan (Essentials or Enterprise).
+
+### Unit Test: Populate JWT From FusionAuth
+
+In this final unit test, let's look at how to check user information available in FusionAuth to determine custom fields to return to your app.
+
+There are two objects to consider. The first is the JWT fields that are returned to your app by default when a user logs in.
+
+```js
+{
+  aud: 'e9fdb985-9173-4e01-9d73-ac2d60d1dc8e',
+  exp: 1692273965,
+  iat: 1692270365,
+  iss: 'acme.com',
+  sub: '00000000-0000-0000-0000-111111111111',
+  jti: '47cc65ce-3981-4e40-8c99-5221d69c53da',
+  authenticationType: 'PASSWORD',
+  email: 'richard@example.com',
+  email_verified: true,
+  applicationId: 'e9fdb985-9173-4e01-9d73-ac2d60d1dc8e',
+  roles: [ 'Petty cash keeper', 'First aider' ],
+  auth_time: 1692270365,
+  tid: 'd7d09513-a3f5-401c-9685-34ab6c552453',
+  message: 'goodbye world'
+}
+```
+
+The second object is the user supplied to your `populate()` function in a lambda.
+
+```js
+user: {
```

Comment:
Please pull these from https://github.com/FusionAuth/fusionauth-site/tree/master/site/docs/src/json as those json files will be updated over time.